snuba/cli/cleanup.py

```diff
@@ -51,10 +51,16 @@ def cleanup(
     dataset = get_dataset(dataset_name)
     writable_storage = dataset.get_writable_storage()
     assert writable_storage is not None, "Dataset has no writable storage"
+    conn = writable_storage.get_cluster().get_connection(ClickhouseClientSettings.CLEANUP)
```

Comment:
There are a few changes to clusters today, so please make sure you've merged the latest. What is of concern to this PR is that `get_connection` was renamed to `get_query_connection`.

However I think it would be better to actually not use this method at all and expose the user and password directly from the cluster. 

To me going through the connection to get it is adds unnecessary indirection since the user/password is actually a property of the cluster. And it establishes a connection with ClickHouse that we don't actually need to use.

I would suggest adding a method to clusters `get_credentials()` that returns `(user, password)`.

Then here we can use it like:
`user, password = writable_storage.get_cluster().get_credentials()`

The optimize script would be exactly the same.

What do you think?

Reply:
Thank you, sounds good to me. Fixing it.

Reply:
Hi @lynnagara I have updated the PR as per your suggestion. Thank you for the reviews.